src/lib/trust_system/record.ml

```diff
@@ -43,10 +43,10 @@ end) : S = struct
      records, and is not exposed outside this module. *)
   let update {trust; trust_last_updated; banned_until_opt} =
     let now = Now.now () in
-    let elapsed_time = Time.diff now trust_last_updated in
-    (* If time is non-monotonic lots of stuff is broken, so I think it's fine
-       not to do any error handling here. See #1494. *)
-    assert (Time.Span.is_non_negative elapsed_time) ;
+    let elap = Time.diff now trust_last_updated in
+    let elapsed_time =
```

Comment:
would be good to have a comment that explains why we might see a negative elapsed time